### PR TITLE
Fix the position of the TOS on the sign on page to be above the keyboard

### PR DESCRIPTION
**Pull request by tgolen**
### Fixed Issues
$ https://github.com/Expensify/App/issues/11277   

### Tests
Note: We decided in the GH issue that mobile web platforms simply cannot be supported with the keyboard avoiding behavior. For mobile web platforms, as long as the user can tap outside of the input to make it lose focus, and they can scroll the page to view the content behind the keyboard, then that is OK 👍 

1. Sign out
2. Tap or click on the email input element on the sign-in form to give it focus
3. (on mobile platforms) Verify the keyboard comes up
4. (on mobile native, NOT mobile web platforms) Verify the TOS is visible above the keyboard
4. (on mobile web, NOT mobile native platforms) Verify the page can be scrolled to view the TOS link
5. Click or tap outside of the input so that it loses focus
6. (on mobile platforms) Verify the keyboard is dismissed
7. Enter an email and click next
8. Repeat steps 2-6 on the password form

- [x] Verify that no errors appear in the JS console

### Offline tests
Same as above.

Note: For offline mode, there is the "offline indicator" which adds extra items to the vertical height of the page. On the password form, this means that it pushes the TOS behind the keyboard again. This is OK :+1:

<details>
<summary>Offline Screenshots</summary>

image

image

</details>

### QA Steps
Same as above

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/1228807/207378136-f968d6ec-deb5-4370-a07d-ed3d40aaf463.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/1228807/207378172-fdb40ae3-aaff-40bc-8bb1-09ba66970fc2.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/1228807/207378210-dd8c8b75-2e00-482c-972d-343f0387e423.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/1228807/207378229-33c1d91c-489c-46e2-9f58-b4eb70c65187.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/1228807/207378278-3ee65a5a-5b76-4c2a-b98e-64876b70a3fe.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/1228807/207378318-63102f3f-c22e-4910-b029-b46a240c8227.mp4

</details>

### CODE REVIEW

**Comment by jasperhuangg** [comment]
Hey @tgolen!

To try to get something with Mobile Safari, I experimented with the `flex-shrink` property, but since it's just an inverse of `flex-grow`, it suffers from the same problems as any other flex solution. 

The Mobile Safari keyboard incorrectly adds extra space at the bottom of the screen, which doesn't affect the flex of any of the elements above it:

Screen Shot 2022-12-13 at 4 25 32 PM

I didn't spend too long on this as Shawn mentioned that the design of this page could change very soon.

**Comment by jasperhuangg** [comment]
The good news is that I was able to figure out some of the style issues you were experiencing on iOS Native. Instead of conditionally adding `marginBottom` to the `View` encompassing the `TermsAndLicenses` component, I simply pass that margin into the `keyboardVerticalOffset` prop. Everything seems to work great with this:


https://user-images.githubusercontent.com/31285285/207264742-1ad4b200-423c-4061-9ab4-5f0631204cb2.mp4

Android still works exactly the same as before with this change:

https://user-images.githubusercontent.com/31285285/207264909-fa26a827-9d99-4314-a93f-af4d3e51b09c.mov

And the behavior on all other touchscreen keyboard platforms is unaffected.

**Comment by tgolen** [comment]
Hey, that's a great find, thank you! I didn't know about that prop. It makes sense that Android wouldn't change because the KAV is a no-op for Android (it's just an empty fragment). I'll get this cleaned up today and ready for review.

File: src/pages/signin/SignInPageLayout/SignInPageContent.js [hunk 1]
```diff
@@ -46,7 +47,13 @@ const SignInPageContent = (props) => {
                     !props.isSmallScreenWidth && styles.signInPageLeftContainerWide,
                 ]}
             >
-                <View style={[styles.flex1, styles.alignSelfCenter, styles.signInPageWelcomeFormContainer]}>
+                <KeyboardAvoidingView
+                    behavior="padding"
```
Comment (tgolen) [inline src/pages/signin/SignInPageLayout/SignInPageContent.js:51]: The linter is complaining about the indentation on this line... but the linter passes locally OK, and this definitely looks like the correct indentation to me.

```
/home/runner/work/App/App/src/pages/signin/SignInPageLayout/SignInPageContent.js
Error:   51:21  error  Expected indentation of 16 space characters but found 20  react/jsx-indent
```

@jasperhuangg can you spot the problem by any chance?

**Review by sobitneupane — COMMENTED** [review]
When offline some part of footer is cut-off in password page in Android Native.


https://user-images.githubusercontent.com/25876548/207538554-af31303f-7e1d-4f52-86c3-1cbd5994c298.mov

**Comment by jasperhuangg** [comment]
@sobitneupane 

I think that's okay, there's simply too many elements on the screen for it to fit, and the user can always scroll down a bit to view the rest of the text.

We run into similar issues on the password page on other platforms, and we've decided that it isn't worth fixing.

**Comment by sobitneupane** [comment]
### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/25876548/207539494-1c9a6e7a-c4cf-483c-8111-744760ff0e35.mov


</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://user-images.githubusercontent.com/25876548/207539852-ef0920d3-a531-4521-bc13-be02c4d615d3.mov


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/25876548/207539933-c734b81f-6f21-4ca0-ba0d-c24cf73d8f95.mov



</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/25876548/207540006-130219f9-2b21-42d3-ac7c-37d02bdfb071.mov



</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/25876548/207540071-6093db1f-8621-4975-8c9a-46d8f179dc93.mov


</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/25876548/207539661-3a55cab0-cd17-40ea-b6b1-1034ed29d9f7.mov


</details>

**Comment by sobitneupane** [comment]
Lint error looks unrelated to this PR.

Merging with main might solve the issue according to https://expensify.slack.com/archives/C01GTK53T8Q/p1670954885116019

**Comment by jasperhuangg** [comment]
@sobitneupane ahh beat me to it, I had already merged main before 😅

**Review by jasperhuangg — APPROVED** [review]

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Pull request merged**